### ISSUE TITLE
Project maintenance: add branch protection rules for specific branches

### ISSUE THREAD

**Issue by paulo-ferraz-oliveira** [erlef/otp_builds#22]
Figure out which ones we're gonna end up using and protect those...

**Comment by paulo-ferraz-oliveira** [comment]
Should wait on #1 to understand what is feasible/reasonable.

**Comment by paulo-ferraz-oliveira** [comment]
There's no actions to run, but I'll still just do the following (Settings)

<img width="710" alt="image" src="https://github.com/user-attachments/assets/e6c05425-693f-40da-9ce9-dc8c2484a144">

If this is too strict we can easily revert it. Closing...

**Comment by wojtekmach** [comment]
Could you revert this change? This caused this failure: https://github.com/erlef/otp_builds/actions/runs/11788301224/job/32835157101

**Comment by paulo-ferraz-oliveira** [comment]
Is this using a bot? Or a PAT?

**Comment by paulo-ferraz-oliveira** [comment]
It should be possible to allow bypassing for specific actors, but I'm not sure (yet?) how this is set up. Not having branch protection rules means you can accidentally push to the default branch, for example.

**Comment by paulo-ferraz-oliveira** [comment]
@starbelly, could we create an Actions Bot, for ErlEF to use in repos where automation is required?

**Comment by paulo-ferraz-oliveira** [comment]
I found we had an unused secret (in the settings; it's still there). I tried to use it, but it didn't work (I also don't know how it was generated). I know how to have branch protection rules alongside automation, but I've only seen it work with either a PAT (less recommended option, since it's tied to a person) or a bot (which we could create for ErlEF and use in other repositories).

In any case, for the time being, the rule is Disabled, so the workflow is back to working. Sorry for the noise, @wojtekmach.

For https://github.com/jelly-beam/otp-macos this works because we go through a pull request (automated creation and merge), but I can understand it being noisy / undesired.

**Comment by paulo-ferraz-oliveira** [comment]
If you could help ErlEf in creating a bot for automated actions it'd be nice. It'd simplify the actions, for example, since we wouldn't need `permissions`, but it's not necessary for the now; it's just a nice-to-have, if you have time.
